### PR TITLE
[DevSpaces]: Add targetContainerHostApiServerFqdn to controller resource

### PR DESCRIPTION
**Pull request by sanar-microsoft**
…rce model

(cherry picked from commit 506f8d3e49495d8b394f9a663a8f5c595da64048)

### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Review by mmyyrroonn — APPROVED** [review]
LGTM.

File: specification/devspaces/resource-manager/Microsoft.DevSpaces/stable/2019-04-01/devspaces.json [hunk 1]
```diff
@@ -701,6 +706,9 @@
     },
     "OrchestratorSpecificConnectionDetails": {
       "description": "Base class for types that supply values used to connect to container orchestrators",
+      "required": [
+        "instanceType"
+      ],
```
Comment (sanjaiganesh) [inline specification/devspaces/resource-manager/Microsoft.DevSpaces/stable/2019-04-01/devspaces.json]: this is breaking change.. 
[https://microsoft.sharepoint.com/:w:/r/teams/azureresourcemanagerteam/_layouts/15/Doc.aspx?sourcedoc=%7B66505FCE-8057-4524-B84F-4CEFE2D8A77C%7D&file=ARMResourcesWithStrongIdentity.docx&action=default&mobileredirect=true](https://microsoft.sharepoint.com/:w:/r/teams/azureresourcemanagerteam/_layouts/15/Doc.aspx?sourcedoc=%7B66505FCE-8057-4524-B84F-4CEFE2D8A77C%7D&file=ARMResourcesWithStrongIdentity.docx&action=default&mobileredirect=true)

Comment (sanar-microsoft) [reply specification/devspaces/resource-manager/Microsoft.DevSpaces/stable/2019-04-01/devspaces.json]: I'll revert this. This change is not needed.

Comment (sanar-microsoft) [reply specification/devspaces/resource-manager/Microsoft.DevSpaces/stable/2019-04-01/devspaces.json]: @sanjaiganesh I've updated the swagger spec, please take a look.

**Comment by MyronFanQiu** [comment]
@sanar-microsoft Hello. Could you solve the CI errors and I can merge this PR.

**Comment by sanar-microsoft** [comment]
@MyronFanQiu I got an exception for suppressing the semantic error from Ray:
    ```
code: 'SEMANTIC_VALIDATION_ERROR',
    id: 'OAV119',
    innerError:
	code: 'OAV131',
        error: 'DISCRIMINATOR_NOT_REQUIRED'

```

I'll get this fixed in the next API version of the RP.

**Comment by MyronFanQiu** [comment]
@sanar-microsoft Sure. You can add the suppression in this PR.

**Comment by sanar-microsoft** [comment]
@MyronFanQiu I added suppression, but apparently that way is not correct. Can you let me know how to add/fix it? Also, how to run prettier check locally?

**Comment by MyronFanQiu** [comment]
@sanar-microsoft about prettiercheck you can follow this instrucment.
```
https://dev.azure.com/azure-sdk/public/_build/results?buildId=253850&view=logs&j=b56e9678-4dad-5642-289c-5530cc4dad11&t=9e18c5d1-dc43-58a1-0344-6ef37d05bd7f&l=34
```

**Comment by MyronFanQiu** [comment]
@raych1 Hello. Could you help check the suppression? Meanwhile, is there a doc to teach us how to add suppression?

**Comment by raych1** [comment]
> @raych1 Hello. Could you help check the suppression? Meanwhile, is there a doc to teach us how to add suppression?

@sanar-microsoft , please refer to this [readme](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/datafactory/resource-manager/readme.md) and search for `oav131` in page. // CC @MyronFanQiu

**Comment by sanar-microsoft** [comment]
@raych1 I added it to the readme, but the same check failed. Is there something missing here?

**Comment by raych1** [comment]
@sanar-microsoft , try this:
## Suppression

``` yaml
directive:
  - from: devspaces.json
    suppress: OAV131 
    reason: Testing purpose.
```

File: specification/devspaces/resource-manager/readme.md [hunk 2]
```diff
@@ -135,3 +135,10 @@ uncomment the  `exclude-file` section below and add the file paths.
 #  - $(this-folder)/Microsoft.Example/stable/2010-01-01/somefile.json
 ```
 
+## Suppressions
+``` yaml
+directive:
```
Comment (sanar-microsoft) [inline specification/devspaces/resource-manager/readme.md]: @raych1 Apparently this didn't work either

**Comment by sanar-microsoft** [comment]
@MyronFanQiu I've fixed the suppression, please take a look. The issue was due to a typo, "Suppressions" vs "Suppression"

@raych1 Thank you for the help

**Pull request merged**